lib/plugins/aws/provider/awsProvider.js

```diff
@@ -328,20 +422,45 @@ class AwsProvider {
                 },
               },
             },
+            memorySize: { $ref: '#/definitions/awsLambdaMemorySize' },
+            package: { $ref: '#/definitions/awsLambdaPackage' },
             resourcePolicy: {
               type: 'array',
               items: {
                 type: 'object',
               },
             },
+            role: { $ref: '#/definitions/awsLambdaRole' },
+            runtime: { $ref: '#/definitions/awsLambdaRuntime' },
+            tags: { $ref: '#/definitions/awsResourceTags' },
+            timeout: { $ref: '#/definitions/awsLambdaTimeout' },
+            tracing: {
+              type: 'object',
+              properties: {
+                lambda: { $ref: '#/definitions/awsLambdaTracing' },
+                apiGateway: { type: 'boolean' },
+              },
+              additionalProperties: false,
+            },
+            vpc: { $ref: '#/definitions/awsLambdaVpcConfig' },
+            versionFunctions: { $ref: '#/definitions/awsLambdaVersionning' },
             websocketsApiName: { type: 'string' },
             websocketsApiRouteSelectionExpression: { type: 'string' },
-            apiGateway: { type: 'object', properties: { websocketApiId: { type: 'string' } } },
           },
         },
         function: {
-          // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8017
           properties: {
+            awsKmsKeyArn: {
+              oneOf: [
+                { $ref: '#/definitions/awsKmsArnString' },
+                { $ref: '#/definitions/awsCfFunction' },
+              ],
+            },
```

Comment:
I think defintion for `service. awsKmsKeyArn` and `functions[].awsKmsKeyArn` should be same.

I understand that's it's now problematic, as we need a definition for `awsCfFunction` in top level (or used relaxed form). I would either allow it as `{ type: 'object' }` there, or move all needed AWS definitions top level (they'll be under this TODO comment, to be moved back to AWS provider in v3)

Reply:
OK, I put `{ type: 'object' }` at root level with a comment (I don't see the interest of moving all cfFunctions definition awaiting for this key to be moved to `provider` level).